alpa/device_mesh.py

```diff
@@ -1146,9 +1242,9 @@ def shard_args_to_arrays(self, avals: Sequence[ShapedArray],
                              args: Sequence[np.array]):
         arrays = []
         for i in range(len(avals)):
-            buffers = _shard_array(args[i], self, shard_indices[i])
+            remote_ref = _shard_array(args[i], self, shard_indices[i])[0]
```

Comment:
This extra `[0]` seems wired. Any way to eliminate it?

Reply:
We used to have some `if` to return a single value instead of a list when the input is not iterable or some arg is 1. but this feature makes some code more complicated now.
For example, for the case that `num_arg == 1`, we need to add code like `ret = np.array([ret])`. Previously we do not have the problem because when `num_arg == 1`, if `num_devices != 1`(this is a usual case), the special branch will not be hit.
I didn't find a perfect balance between the `[0]` and `if num == 1: ret = [ret]`. Do you have some suggestions?